### PR TITLE
Set context actual domain id

### PR DESCRIPTION
**Pull request by ivanpauno**
See https://github.com/ros2/rcl/pull/718.

### CODE REVIEW

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 1]
```diff
@@ -1154,6 +1154,8 @@ extern "C" rmw_ret_t rmw_init(const rmw_init_options_t * options, rmw_context_t
 
   context->instance_id = options->instance_id;
   context->implementation_identifier = eclipse_cyclonedds_identifier;
+  context->actual_domain_id =
+    RMW_DEFAULT_DOMAIN_ID != options->domain_id ? options->domain_id : 0u;
```
Comment (hidmic) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:1158]: @ivanpauno [this line](https://github.com/ros2/rmw_cyclonedds/pull/208/files#diff-d03c99cafce317e1a93de46e787e0235R944) has to change too.

Comment (ivanpauno) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1158]: @hidmic can you clarify which line you're referring to?
The hyperlink redirects to this same comment.

Comment (ivanpauno) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1157]: Ok, I think I guessed which line was: https://github.com/ros2/rmw_cyclonedds/pull/208/commits/a8b45bfd2ccf0639bf769506422e3e38b63dcec3

Comment (hidmic) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1157]: Uh, that's so weird. It's almost like if Github's line hyperlinks within a PR diff are broken. Anyhow, you guessed right :)

Comment (ivanpauno) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1157]: > It's almost like if Github's line hyperlinks within a PR diff are broken

Yeah, links within a PR diff are broken in my experience after the review is sent.
Opening the file and getting the permalink always works correctly.

File: rmw_cyclonedds_cpp/src/rmw_node.cpp [hunk 2]
```diff
@@ -1147,13 +1145,14 @@ extern "C" rmw_ret_t rmw_init(const rmw_init_options_t * options, rmw_context_t
     return RMW_RET_INVALID_ARGUMENT;
   }
 
-  const rmw_context_t zero_context = rmw_get_zero_initialized_context();
-  assert(0 == std::memcmp(context, &zero_context, sizeof(rmw_context_t)));
```
Comment (hidmic) [inline rmw_cyclonedds_cpp/src/rmw_node.cpp:1151]: @ivanpauno unrelated?

Comment (ivanpauno) [reply rmw_cyclonedds_cpp/src/rmw_node.cpp:1151]: See https://github.com/ros2/rmw_connext/pull/443#discussion_r458314262

**Pull request merged**